executor/distsql.go

```diff
@@ -846,14 +855,27 @@ func (w *tableWorker) pickAndExecTask(ctx context.Context) {
 	}
 }
 
-func (e *IndexLookUpExecutor) getHandle(row chunk.Row, handleIdx []int, isCommonHandle bool) (handle kv.Handle, err error) {
+func (e *IndexLookUpExecutor) getHandle(row chunk.Row, handleIdx []int,
+	isCommonHandle bool, tp getHandleType) (handle kv.Handle, err error) {
 	if isCommonHandle {
 		var handleEncoded []byte
 		var datums []types.Datum
 		for i, idx := range handleIdx {
+			// If the new collation is enabled and the handle contains non-binary string,
+			// we change the collation to "binary" to avoid encode sortKey again.
```

Comment:
```suggestion
			// we change the collation to "binary" to avoid encode sortKey again.
```
```
// If the new collation is enabled and the handle contains non-binary string, the handle in the index is encoded as "sortKey". So we cannot restore its original value(the primary key) here. 
// we use a trick to avoid encoding the "sortKey" again by changing the charset collation to `binary`.
// TODO: Add the restore value to the secondary index to remove this trick.